File: backend/app/models/comment.rb
class Comment < ApplicationRecord
  belongs_to :user
  belongs_to :post
  belongs_to :parent_comment, class_name: 'Comment', optional: true
  has_many :replies, class_name: 'Comment', foreign_key: 'parent_comment_id', dependent: :destroy

  validates :content, presence: true, length: { maximum: 255 }

  # トップレベルコメント（リプライではない）
  scope :top_level, -> { where(parent_comment_id: nil) }
  
  # リプライかどうか判定
  def reply?
    parent_comment_id.present?
  end

  # リプライの数を取得
  def replies_count
    replies.count
  end
end

File: frontend/src/types/comment.ts
// コメント関連の型定義
import { ID, DateTimeString } from './common'

/**
 * コメント投稿者情報
 */
export interface CommentUser {
  id: ID
  name: string
}

/**
 * 返信先情報
 */
export interface ReplyingTo {
  id: ID
  user_name: string
}

/**
 * コメント型
 */
export interface Comment {
  id: ID
  content: string
  created_at: DateTimeString
  parent_comment_id?: ID
  user: CommentUser
  replying_to?: ReplyingTo
  replies?: Comment[]
}
